build.sh

```diff
@@ -35,43 +62,41 @@ build_type=RelWithDebInfo
 #extra_enable="-D HAVE_PARPORT=1"
 extra_enable=""
 
-build_flags=""
-
 case "${ostype}" in
     linux)
-	# try to find out whether this is an Embedded Linux
-	# platform (e.g. Raspberry Pi)
-	machine=$(uname -m)
-	if expr "${machine}" : '^\(arm\|aarch\)' >/dev/null
-	then
-	    extra_enable="${extra_enable} -D HAVE_LINUXGPIO=1 -D HAVE_LINUXSPI=1"
-	fi
-	;;
+        # try to find out whether this is an Embedded Linux
+        # platform (e.g. Raspberry Pi)
+        machine=$(uname -m)
+        if expr "${machine}" : '^\(arm\|aarch\)' >/dev/null
+        then
+            extra_enable="${extra_enable} -D HAVE_LINUXGPIO=1 -D HAVE_LINUXSPI=1"
+        fi
+        ;;
 
     darwin)
-	# determine whether we are running using Mac Ports
-	# if not, assume Mac Brew
-	if [ -f /opt/local/bin/port ]
-	then
-	    build_flags="${build_flags} -D CMAKE_C_FLAGS=-I/opt/local/include -D CMAKE_EXE_LINKER_FLAGS=-L/opt/local/lib"
-	else
+        # determine whether we are running using Mac Ports
+        # if not, assume Mac Brew
+        if [ -f /opt/local/bin/port ]
+        then
+            build_flags="${build_flags} -D CMAKE_C_FLAGS=-I/opt/local/include -D CMAKE_EXE_LINKER_FLAGS=-L/opt/local/lib"
```

Comment:
It is customary that user provided flags on the command line overwrite default settings of scripts. Here it looks like it is the other way round. If the user were to set build_flags on the command line on a Mac using Brew then the script would overwrite the user choice here.

Unless there is a good reason for your choice, please ensure that the CL options overwrite the script options irrespective of chosen OS.

Reply:
I was too quick in my review: the rhs of the assignment actually prepends the user build_flags to any ones that might be added here. So all good on that front and no further need to address this comment.